### PR TITLE
Bump CSI Spec to 1.0.0 and gRPC to 1.13.0

### PR DESCRIPTION
**Pull request by saad-ali**
**What type of PR is this?**
> Uncomment only one, leave it on its own line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
/kind feature
> /kind flake

**What this PR does / why we need it**:
This PR picks up CSI spec [v1.0.0](https://github.com/container-storage-interface/spec/releases/tag/v1.0.0-rc1)

It also bumps the gRPC version since that is out of date.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://github.com/kubernetes/enhancements/issues/178

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Kubernetes v1.13 moves support for Container Storage Interface to GA. As part of this move Kubernetes now supports CSI v1.0.0 and drops support for CSI 0.3 and older releases. Older CSI drivers must be updated to CSI 1.0 in order to work with Kubernetes 1.13+.
```

### CODE REVIEW

**Comment by saad-ali** [comment]
CSI Tests will fail until the drivers have been updated to pick up 1.0.0.

**Comment by AishSundar** [comment]
@saad-ali how much delta time window are we looking between bumping CSI Spec to 1.0.0 and updating the drivers? Could you please ping this issue when everything is updated and CSI tests are expected to pass again?

@jberkus @mariantalla @mortent as FYI on upcoming CSI test failures in  CI signal

**Comment by msau42** [comment]
FYI csi tests are run as part of pull now so we cannot merge this PR until all CI drivers are updated.  Also we will run the non pull csi tests (serial, slow, alpha) to minimize the possibility of test breakage post merge.

**Comment by saad-ali** [comment]
/assign @thockin
To unblock approvals while we work on the CSI E2E tests fixes

**Comment by saad-ali** [comment]
/hold
Until tests are green

**Comment by saad-ali** [comment]
All external components have been updated. I've updated this PR pointing to canary builds of the external components for tests.

Initial tests results are looking good. All except one `csi-hostpath` tests passed:
```
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSNov 14 21:59:11.240: INFO: Running AfterSuite actions on all nodes
Nov 14 21:59:11.240: INFO: Running AfterSuite actions on node 1


Summarizing 1 Failure:

[Fail] [sig-storage] CSI Volumes [Driver: csi-hostpath] [Testpattern: Dynamic PV (block volmode)] volumeMode[Feature:BlockVolume] [It] should fail in binding dynamic provisioned PV to PVC 
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/storage/testsuites/volumemode.go:333

Ran 19 of 1909 Specs in 2781.536 seconds
FAIL! -- 18 Passed | 1 Failed | 0 Pending | 1890 Skipped --- FAIL: TestE2E (2781.61s)
FAIL
```

CSI tests for GCE PD are running on my test cluster, so far they're green. Will let them run over night.

Next steps:
* 11:20 AM PT CSI 1.0.0 is planned to be cut.
* I will rebase this PR with that.
* We'll cut official releases of the external components.
* I'll update this PR to point to the official containers instead of canary.
* Merge this PR.

**Review by pohly — CHANGES_REQUESTED** [review]
The changelog entry for this PR is empty. Does that mean that there really are no user-visible changes, in particular that driver deployments of CSI drivers based on CSI spec 0.3.0 and the 0.3.x sidecar containers will continue to work?

If so, then why the "CSI Tests will fail until the drivers have been updated to pick up 1.0.0." comment and the concerted effort to get everything updated?

It rather looks like this is a major change that will break older drivers. I understand that CSI is still in the "move fast and break things" phase, but at least communicate that by:
- adding a big, fat warning in the changelog
- updating https://github.com/kubernetes-csi/docs *before* 1.13 gets released

Kubernetes 1.12 also broke existing drivers, and that surprised and/or confused people because they had to figure out the root cause of failures without clear guidance.

**Comment by saad-ali** [comment]
The GCE PD tests I ran last night, also look good:

```
Summarizing 1 Failure:

[Fail] [sig-storage] CSI Volumes [Driver: com.google.csi.gcepd][Serial] [Testpattern: Dynamic PV (block volmode)] volumeMode[Feature:BlockVolume] [It] should fail in binding dynamic provisioned PV to PVC 
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/storage/testsuites/volumemode.go:333

Ran 23 of 1909 Specs in 4093.133 seconds
FAIL! -- 22 Passed | 1 Failed | 0 Pending | 1886 Skipped --- FAIL: TestE2E (4093.25s)
FAIL
```

Again the 1 failure is expected since that is a test for an alpha feature disabled by default. Running the alpha feature tests to sanity check everything.

@pohly Yes, I will update the release note section.

**Comment by AishSundar** [comment]
@marpaia to followup on the release notes update mentioned in https://github.com/kubernetes/kubernetes/pull/71020#pullrequestreview-175296511

@saad-ali just cross checking if we should hold off on this bump until https://github.com/kubernetes/kubernetes/pull/71020#pullrequestreview-175296511 is addressed?

**Comment by saad-ali** [comment]
> @saad-ali just cross checking if we should hold off on this bump until #71020 (review) is addressed?

Nope we can do that in parallel. Release notes and these changes will be released together as part of 1.13.

File: test/e2e/testing-manifests/storage-csi/gce-pd/node_ds.yaml
Comment (davidz627) [inline test/e2e/testing-manifests/storage-csi/gce-pd/node_ds.yaml]: this whole file should be updated to pull in the newest yamls:
https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/master/deploy/kubernetes/base/node.yaml

File: test/e2e/testing-manifests/storage-csi/gce-pd/controller_ss.yaml
Comment (davidz627) [inline test/e2e/testing-manifests/storage-csi/gce-pd/controller_ss.yaml]: make sure we're using the newest yamls here:
https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/master/deploy/kubernetes/base/controller.yaml

File: test/e2e/testing-manifests/storage-csi/gce-pd/controller_ss.yaml
Comment (davidz627) [inline test/e2e/testing-manifests/storage-csi/gce-pd/controller_ss.yaml]: update version when prod release cut

**Review by davidz627 — COMMENTED** [review]
also update the GCE PD Driver names in the storage class for the e2e tests

File: test/e2e/testing-manifests/storage-csi/gce-pd/node_ds.yaml
Comment (davidz627) [reply test/e2e/testing-manifests/storage-csi/gce-pd/node_ds.yaml]: specifically there are some updates around prestop lifecycle and possibly udev rules

Comment (saad-ali) [reply test/e2e/testing-manifests/storage-csi/gce-pd/node_ds.yaml]: Done.

File: test/e2e/testing-manifests/storage-csi/gce-pd/controller_ss.yaml
Comment (saad-ali) [reply test/e2e/testing-manifests/storage-csi/gce-pd/controller_ss.yaml]: Done

File: test/e2e/testing-manifests/storage-csi/gce-pd/controller_ss.yaml
Comment (saad-ali) [reply test/e2e/testing-manifests/storage-csi/gce-pd/controller_ss.yaml]: Yep.

File: test/e2e/testing-manifests/storage-csi/hostpath/hostpath/csi-hostpathplugin.yaml
Comment (davidz627) [inline test/e2e/testing-manifests/storage-csi/hostpath/hostpath/csi-hostpathplugin.yaml]: I think this line: https://github.com/kubernetes/kubernetes/blob/519eee70ed7fbdaa41dc730c644ca019e508ab86/test/e2e/testing-manifests/storage-csi/hostpath/hostpath/csi-hostpathplugin.yaml#L14

might need to get changed to `csi-node-sa` based on the other changes

Comment (saad-ali) [reply test/e2e/testing-manifests/storage-csi/hostpath/hostpath/csi-hostpathplugin.yaml]: Fixed.

**Comment by saad-ali** [comment]
No other PR, I will just push an update to this PR pointing to the new containers. Should be done in the next hour or two.

**Comment by saad-ali** [comment]
Ok, all good to go.

**Pull request merged**
